src/Compilers/Test/Utilities/VisualBasic/CompilationTestUtils.vb

```diff
@@ -923,18 +926,18 @@ Friend Module CompilationUtils
         Dim expectedReader = New StringReader(expected)
         Dim actualReader = New StringReader(actual)
 
-        Dim expectedBuilder As New StringBuilder()
-        Dim actualBuilder As New StringBuilder()
+        Dim expectedBuilder = _PooledStringBuilderPool.Allocate()
+        Dim actualBuilder = _PooledStringBuilderPool.Allocate()
         Dim expectedLine = expectedReader.ReadLine()
         Dim actualLine = actualReader.ReadLine()
 
         While expectedLine IsNot Nothing AndAlso actualLine IsNot Nothing
             If Not expectedLine.Equals(actualLine) Then
-                expectedBuilder.AppendLine("<! " & expectedLine)
-                actualBuilder.AppendLine("!> " & actualLine)
+                expectedBuilder.Builder.AppendLine("<! " & expectedLine)
```

Comment:
This discussion thread should be followed up on. Here is a link for quick navigation https://github.com/dotnet/roslyn/pull/17134#discussion_r102540128.